magicctapipe/reco/estimators.py

```diff
@@ -55,76 +53,81 @@ def __init__(
 
     def fit(self, event_data):
         """
-        Trains RFs per telescope with input training samples.
+        Train a RF per telescope.
 
         Parameters
         ----------
         event_data: pandas.core.frame.DataFrame
-            Pandas data frame of training samples
+            Data frame of shower events
         """
 
         self.telescope_rfs.clear()
 
-        event_data.dropna(subset=self.features, inplace=True)
-
-        if self.use_unsigned_features:
-            event_data[self.features] = np.abs(event_data[self.features])
-
-        # Train RFs per telescope:
-        tel_ids = np.unique(event_data.index.get_level_values("tel_id"))
+        # Train a RF per telescope
+        tel_ids = np.unique(event_data["tel_id"])
 
         for tel_id in tel_ids:
 
-            df_events = event_data.query(f"tel_id == {tel_id}")
+            df_events = event_data.query(f"tel_id == {tel_id}").copy()
+            df_events.dropna(subset=self.features, inplace=True)
+
+            if self.use_unsigned_features:
```

Comment:
you could write it slightly more compact 
```
x_train = df_events[self.features].to_numpy()
if self.use_unsigned_features:
    x_train = np.abs(x_train)
```

Reply:
Thanks, I changed it as you suggested.